src/udev/meson.build

```diff
@@ -124,12 +124,30 @@ libudev_basic = static_library(
         c_args : ['-fvisibility=default'])
 
 libudev_static = static_library(
-        'udev',
+        'udev_static',
         'udev.h',
         include_directories : includes,
         link_with : udev_link_with,
         link_whole : libudev_basic)
 
+static = get_option('static-libudev')
+if static != 'false'
+    if static == 'true' or static == 'pic'
+        pic = true
+    else
+        pic = false
+    endif
+    install_libudev_static = static_library(
+        'udev',
+        'udev.h',
+        include_directories : includes,
+        link_with : udev_link_with,
+        link_whole : libudev_basic,
+        install : true,
+        install_dir : rootlibdir,
+        pic : pic)
+endif
```

Comment:
I'd like this simplified by removing the conditional (here and for the other one too):
```python
static_libudev = get_option('static-libudev')
install_libudev_static = static_library(
        'udev',
        'udev.h',
        include_directories : includes,
        link_with : udev_link_with,
        link_whole : libudev_basic,
        build_by_default : static_libudev != 'false',
        install : static_libudev != 'false',
        install_dir : rootlibdir,
        pic : static_libudev == 'true' or static == 'pic')
```
This has the advantage that the static library can be build by explicit call
```console
$ ninja -C build src/udev/libudev.a
ninja: Entering directory `build'
[1/1] Linking static target src/udev/libudev.a.
```
We do that with man pages, which can be built even if `-Dman=false`, and its useful for testing and debugging and such.